File: engine/b/src/main/java/net/crazyinvestor/engine_b/repository/QuoteRepository.java
package net.crazyinvestor.engine_b.repository;

import net.crazyinvestor.engine_b.entity.Quote;
import net.crazyinvestor.engine_b.entity.Symbol;
import net.crazyinvestor.engine_b.entity.Ticker;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface QuoteRepository extends JpaRepository<Quote, Long> {
    Optional<Quote> findBySymbol(Symbol symbol);
}

File: engine/a/ticker_producer_engine/src/main/kotlin/net/crazyinvestor/ticker_producer_engine/repository/QuoteRepository.kt
package net.crazyinvestor.ticker_producer_engine.repository;

import net.crazyinvestor.ticker_producer_engine.entity.Quote
import net.crazyinvestor.ticker_producer_engine.entity.Symbol
import org.springframework.stereotype.Repository

import java.util.Optional;

@Repository
class QuoteRepository {
    fun findBySymbol(symbol: Symbol): Optional<Quote> {
        println("MOCK REPOSITORY2 $symbol")

        return Optional.of(Quote(1L, symbol, null, null, symbol.quote?.updatedAt))
    }

    fun save(quote: Quote): Quote {
        println("save MOCK REPOSITORY2 $quote")
        return quote
    }
}
